cli/flox/src/commands/pull.rs

```diff
@@ -274,11 +286,19 @@ impl Pull {
         };
         // endregion
 
-        // The pulled generation already has a lock,
-        // so we can skip locking.
-        let result = env
-            .build(flox)
-            .and_then(|store_paths| env.link(&store_paths));
+        let result = if env.generations_metadata()?.current_gen() != generation
+            && let Some(generation) = generation
```

Comment:
nit: we could avoid reading metadata if no generation is requested in the first place.

```suggestion
        let result = if let Some(generation) = generation 
            && env.generations_metadata()?.current_gen() != generation
```